slackroll

```diff
@@ -455,17 +455,34 @@ def get_blacklist(): # Get blacklist expression list
     return []
 
 def get_blacklist_re(): # Almost the same, but already converted to regular expressions
+    regex_list = []
     try:
-        return [re.compile(x) for x in get_blacklist()]
+        for x in get_blacklist():
+            data = x.split('@', 1)
+            if len(data) > 1:
+                pkg_regex, url_regex = data
+                regex_list.append((re.compile(pkg_regex), re.compile(url_regex)))
+            else:
+                regex_list.append((re.compile(x), None))
+
+        return regex_list
     except re.error:
         sys.exit('ERROR: invalid regular expression found in blacklist')
 
 def add_blacklist_exprs(expressions): # Add expressions to the blacklist
     valid_ones = []
     for regex_str in expressions:
         try:
-            re.compile(regex_str)
-            valid_ones.append(regex_str)
+            data = regex_str.split('@', 1)
```

Comment:
After splitting, we would always do the "if" block inconditionally.

Reply:
Updated